tests/test_end_to_end.py

```diff
@@ -0,0 +1,57 @@
+"""An end to end example and test."""
+from typing import Callable
+
+from sklearn import datasets, svm
+from sklearn.model_selection import cross_val_score
+
+from kotsu import run
+from kotsu.registration import ModelRegistry, ValidationRegistry
+
+
+model_registry = ModelRegistry()
+
+model_registry.register(
+    id="SCV-v1",
+    entry_point=svm.SVC,
+    kwargs={"kernel": "linear", "C": 1, "random_state": 1},
+)
+
+model_registry.register(
+    id="SCV-v2",
+    entry_point=svm.SVC,
+    kwargs={"kernel": "linear", "C": 0.5, "random_state": 1},
+)
+
+validation_registry = ValidationRegistry()
+
+
+def factory_iris_cross_val(folds: int) -> Callable:
+    """Factory for iris cross validation."""
+
+    def iris_cross_val(model) -> dict:
+        """Iris classification cross validation."""
+        X, y = datasets.load_iris(return_X_y=True)
```

Comment:
see comment on run - I think the load dataset here should potentially be outside of the actual validation call and in the validation setup / factory?